### PR TITLE
feat: add logger factory interface

### PR DESCRIPTION
**Pull request by pgautier404**

### CODE REVIEW

File: README-logging.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+$loggerFactory = new MonologFactory();
+$log = $loggerFactory->getLogger("channel_name");
+```
+
+If desired, your implementation could be extended to allow runtime configuration of stream handlers, formats, and the
+other bells and whistles that `Monolog` and other more fully-featured logging solutions offer.
+
+----------------------------------------------------------------------------------------
+For more info, visit our website at [https://gomomento.com](https://gomomento.com)!
```
Comment (poppoerika) [inline README-logging.md:71]: Maybe we don't need this for logging README?

File: examples/dictionary-example.php [hunk 2]
```diff
@@ -20,20 +18,9 @@
...
-$streamHandler->setFormatter($formatter);
-$logger->pushHandler($streamHandler);
-
-// Or use the built-in minimal logger, equivalent to the above Monolog configuration.
-//$logger = \Momento\Utilities\LoggingHelper::getMinimalLogger();
-// Or discard all log messages.
-//$logger = \Momento\Utilities\LoggingHelper::getNullLogger();
-
-$configuration = Laptop::latest($logger);
+$configuration = Laptop::latest()->withLoggerFactory(new StderrLoggerFactory());
```
Comment (poppoerika) [inline examples/dictionary-example.php:21]: Beautifully abstracted...!

**Review by cprice404 — APPROVED** [review]
❤️

File: README-logging.md (hunk 1 above)
Comment (pgautier404) [reply README-logging.md:71]: I don't think it hurts anything, and we should make it as easy as possible for interested parties to get to our website as possible.

**Review by cprice404 — APPROVED** [review]

**Pull request merged**